File: OleDbConnectionUpdateData/DataOperations.cs
using System;
using System.Data.OleDb;
using System.Threading.Tasks;

namespace OleDbConnectionUpdateData
{
    public class DataOperations
    {
        private static string _connectionString = "TODO";
        
        public delegate void OnFinish();
        public static event OnFinish OnFinishedEvent;
        
        public delegate void OnException(Exception sender);
        public static event OnException OnExceptionEvent;

        public static async Task<bool> PerformUpdateTask()
        {

            return await Task.Run(async () =>
            {
                using (var cn = new OleDbConnection(_connectionString))
                {

                    using (var cmd = new OleDbCommand() { Connection = cn })
                    {

                        cmd.CommandText = "TODO";

                        try
                        {
                            await cn.OpenAsync();
                            var result = await cmd.ExecuteNonQueryAsync();
                            OnFinishedEvent?.Invoke();
                            return result == 1;
                        }
                        catch (Exception ex)
                        {
                            OnExceptionEvent?.Invoke(ex);
                            return false;
                        }
                    }
                }
            
            });
        }

    }
}

File: CheckedListBoxDemo/DataOperations.vb
Imports System.Data.OleDb
Public Class DataOperations
    Private Shared _connectionString As String = "TODO"
    Public Delegate Sub OnFinish()
    Public Shared Event Finished As OnFinish


    Public Delegate Sub OnException(sender As Exception)
    Public Shared Event OnExceptionEvent As OnException

    Public Shared Async Function PerformUpdateTask() As Task(Of Boolean)

        Return Await Task.Run(Async Function()
                                  Using cn = New OleDbConnection(_connectionString)

                                      Using cmd = New OleDbCommand() With {.Connection = cn}

                                          cmd.CommandText = "TODO"

                                          Try
                                              Await cn.OpenAsync()
                                              Dim result = Await cmd.ExecuteNonQueryAsync()
                                              RaiseEvent Finished()
                                          Catch ex As Exception
                                              RaiseEvent OnExceptionEvent(ex)
                                              Return False
                                          End Try
                                      End Using
                                  End Using

                                  Return True

                              End Function)

    End Function




End Class
